text/0156-columnar-encoding.md

```diff
@@ -0,0 +1,1389 @@
+more information on the implementation of this identifier.
+
+The `otlp_arrow_payloads` attribute is a list of `OtlpArrowPayload` messages. Each `OtlpArrowPayload` message represents
+a table of data encoded in a columnar format (metrics, logs, or traces). Although not used in the current version of
+this protocol, this attribute is an array to allow representing different types of entities.
+More details on the `OtlpArrowPayload` columns in the
+section [Mapping OTEL entities to Arrow records](#mapping-otel-entities-to-arrow-records).
+
+The `delivery_type` attribute is an optional attribute that can be used to indicate the type of message delivery
+guarantee expected by the sender. The delivery types are defined by the following protobuf enumeration:
+
+```protobuf
+enum DeliveryType {
+  BEST_EFFORT = 0;
+}
+```
+
+The delivery type `BEST_EFFORT` is the default value. This means that the sender expects the receiver to acknowledge
+receipt of a message and to do its best to pass the message on to the rest of the downstream chain.
+
+More specifically, an `OtlpArrowPayload` protobuf message is defined as:
+
+```protobuf
+message OtlpArrowPayload {
+  // [mandatory] A unique id assigned to a sub-stream of the batch sharing the same schema, and dictionaries.
+  string sub_stream_id = 1;
+
+  // [mandatory] Type of the OTLP Arrow payload.
+  OtlpArrowPayloadType type = 2;
+
+  // [optional] Serialized Arrow dictionaries
+  repeated EncodedData dictionaries = 3;
+
+  // [mandatory] Serialized Arrow Record Batch
+  // For a description of the Arrow IPC format see: https://arrow.apache.org/docs/format/Columnar.html#serialization-and-interprocess-communication-ipc
+  bytes record = 4;
+
+  // [mandatory]
+  CompressionMethod compression = 5;
+}
+
+enum OtlpArrowPayloadType {
+  METRICS = 0;
+  LOGS = 1;
+  SPANS = 2;
+}
+
+message EncodedData {
+  bytes ipc_message = 1;
+  bytes arrow_data = 2;
+}
+
+enum CompressionMethod {
+  NO_COMPRESSION = 0;
+  ZSTD = 1;
+}
+```
+
+The `sub_stream_id` attribute is a unique identifier for a sub-stream of `BatchArrowRecords` sharing the same schema inside
+the
+scope of the current stream. This id will be used receiver side to keep track of the schema and dictionaries for a
+specific
+type of Arrow Records. See the [Substream Id generation](#substream-id-generation) section for more information on the
+implementation
+of this identifier.
+
+The `OtlpArrowPayloadType` enum specifies the `type` of the payload.
+
+The `dictionaries` attribute is a list of binary representations of [Arrow dictionaries](#arrow-dictionary). Several
+dictionaries can be defined in the case of an entity containing several text or binary columns. The dictionaries are
+filled in the first message of their corresponding sub-stream. When a dictionary is updated, the delta dictionary will be
+emitted in the corresponding message within the sub-stream.
+
+The `record` attribute is a binary representation of the Arrow RecordBatch.
```

Comment:
This will be tracked as well with the GH issue already created for one of your previous comment, see https://github.com/f5/otel-arrow-adapter/issues/86

Reply:
This field repeated EncodedData dictionaries = 3; declared in the protobuf spec was in fact not used in the entire project... An easy fix.